server/conn_test.go

```diff
@@ -1171,6 +1172,9 @@ func TestAuthPlugin2(t *testing.T) {
 
 	cc.isUnixSocket = true
 	require.NoError(t, failpoint.Enable("github.com/pingcap/tidb/server/FakeAuthSwitch", "return(1)"))
+	ctx := failpoint.WithHook(context.Background(), func(ctx context.Context, fpname string) bool {
```

Comment:
I think you should remove the `t.Parallel()` in the first line of this function since the test with failpoint shouldn't run parallel.